src/blockchain/index.ts

```diff
@@ -14,23 +18,44 @@ export function ethFactory (): Eth {
   return new Eth(provider)
 }
 
-const CHANNEL_NAME = 'confirmations'
+const CONFIRMATION_CHANNEL = 'confirmations'
+const NEW_BLOCK_EMITTER_CHANNEL = 'new-block'
 
 function channelSetup (app: Application): void {
   if (typeof app.channel !== 'function') {
     // If no real-time functionality has been configured just return
     return
   }
   app.on('connection', (connection: any) => {
-    app.channel(CHANNEL_NAME).join(connection)
+    app.channel(CONFIRMATION_CHANNEL).join(connection)
+    app.channel(NEW_BLOCK_EMITTER_CHANNEL).join(connection)
   })
-  app.service(ServiceAddresses.CONFIRMATIONS).publish(() => app.channel(CHANNEL_NAME))
+  app.service(ServiceAddresses.CONFIRMATIONS).publish(() => app.channel(CONFIRMATION_CHANNEL))
+  app.service(ServiceAddresses.NEW_BLOCK_EMITTER).publish(() => app.channel(NEW_BLOCK_EMITTER_CHANNEL))
 }
 
-export default function (app: Application): void {
+function subscribeAndEmitNewBlocks (app: Application): void {
+  const eth = app.get('eth')
+  const store = getObject()
+  const newBlockEmitter = getNewBlockEmitter(eth)
+  const newBlockEmitterService = app.service(ServiceAddresses.NEW_BLOCK_EMITTER)
+
+  // Subscribe for new blocks
+  newBlockEmitter.on(NEW_BLOCK_EVENT_NAME, (block: BlockHeader) => {
+    logger.debug('New block: ', block)
+    newBlockEmitterService?.emit(NEW_BLOCK_EVENT_NAME, block)
+    store['blockchain.lastFetchedBlockNumber'] = block.number
```

Comment:
Oh I see it now. Why do you emit the whole block header but store only its number? I believe that you should store the whole object so you can `GET` the whole object instead of the only number.

Reply:
Done